File: problems_list/0045.跳跃游戏II/0045-跳跃游戏II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int max_pos=0, end=0, step=0, n=nums.size();
        for(int i=0; i<n-1; i++){
            if(i<=max_pos){
                max_pos=max(max_pos, i+nums[i]);
                if(i==end){
                    end=max_pos;
                    step++;
                }
            }
        }
        return step;
    }
};

File: problems_list/0045.跳跃游戏II/0045-跳跃游戏II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    var step=0, end=0, maxPos=0, n=nums.length;
    for(let i=0; i<n; i++){
        maxPos=Math.max(maxPos, i+nums[i]);
        if(i==n-1) break;
        if(i==end){
            end=maxPos;
            step++;
        }
    }
    return step;
};